src/GraphQL.Tests/Utilities/GetGraphTypeFromTypeTests.cs

```diff
@@ -75,5 +77,127 @@ public void supports_uint_type()
+        [InlineData(typeof(decimal), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(DecimalGraphType))]
+        [InlineData(typeof(float), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(FloatGraphType))]
+        [InlineData(typeof(double), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(FloatGraphType))]
+        [InlineData(typeof(DateTime), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(DateTimeGraphType))]
+        [InlineData(typeof(DateTimeOffset), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(DateTimeOffsetGraphType))]
+        [InlineData(typeof(TimeSpan), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(TimeSpanSecondsGraphType))]
+        [InlineData(typeof(bool), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(BooleanGraphType))]
+        [InlineData(typeof(string), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(StringGraphType))]
+        [InlineData(typeof(Guid), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(IdGraphType))]
+        [InlineData(typeof(Uri), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(UriGraphType))]
+        [InlineData(typeof(MyClass), true, TypeMappingMode.UseBuiltInScalarMappings, null)]
+        [InlineData(typeof(MyEnum), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(EnumerationGraphType<MyEnum>))]
+        //built-in mapping mode - nullable structs
+        [InlineData(typeof(int?), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(IntGraphType))]
+        [InlineData(typeof(decimal?), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(DecimalGraphType))]
+        [InlineData(typeof(Guid?), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(IdGraphType))]
+        [InlineData(typeof(MyEnum?), true, TypeMappingMode.UseBuiltInScalarMappings, typeof(EnumerationGraphType<MyEnum>))]
+        //built-in mapping mode - not nullable types
+        [InlineData(typeof(int), false, TypeMappingMode.UseBuiltInScalarMappings, typeof(NonNullGraphType<IntGraphType>))]
+        [InlineData(typeof(string), false, TypeMappingMode.UseBuiltInScalarMappings, typeof(NonNullGraphType<StringGraphType>))]
+        [InlineData(typeof(decimal), false, TypeMappingMode.UseBuiltInScalarMappings, typeof(NonNullGraphType<DecimalGraphType>))]
+        [InlineData(typeof(Guid), false, TypeMappingMode.UseBuiltInScalarMappings, typeof(NonNullGraphType<IdGraphType>))]
+        [InlineData(typeof(Uri), false, TypeMappingMode.UseBuiltInScalarMappings, typeof(NonNullGraphType<UriGraphType>))]
+        [InlineData(typeof(MyClass), false, TypeMappingMode.UseBuiltInScalarMappings, null)]
+        [InlineData(typeof(MyEnum), false, TypeMappingMode.UseBuiltInScalarMappings, typeof(NonNullGraphType<EnumerationGraphType<MyEnum>>))]
+        //built-in mapping mode - nullable structs
+        [InlineData(typeof(int?), false, TypeMappingMode.UseBuiltInScalarMappings, null)]
```

Comment:
These combinations look strange, but even if we add NRT feature, it will not automatically mean that we can remove `nullable` parameter, because one can have cases when some `int` property **should** map to nullable IntGraphType.